core/src/main/scala/kafka/zk/ZkData.scala

```diff
@@ -757,8 +757,8 @@ case object ExtendedAclChangeStore extends ZkAclChangeStore {
 object ResourceZNode {
   def path(resource: ResourcePattern): String = ZkAclStore(resource.patternType).path(resource.resourceType, resource.name)
 
-  def encode(acls: Set[AclEntry]): Array[Byte] = Json.encodeAsBytes(AclEntry.toJsonCompatibleMap(acls).asJava)
-  def decode(bytes: Array[Byte], stat: Stat): VersionedAcls = VersionedAcls(AclEntry.fromBytes(bytes), stat.getVersion)
+  def encode(acls: Set[AclEntry]): Array[Byte] = Json.encodeAsBytes(AclEntry.toJsonCompatibleMap(acls.asJava))
+  def decode(bytes: Array[Byte], stat: Stat): VersionedAcls = VersionedAcls(immutable.Set().concat(AclEntry.fromBytes(bytes).asScala), stat.getVersion)
```

Comment:
IIRC, `.toSet` can generate the immutable object, so it can be simplified to `AclEntry.fromBytes(bytes).asScala.toSet`

Reply:
Fixed